text/0000-char-uax-31.md

```diff
@@ -0,0 +1,103 @@
+- Feature Name: `char_uax_31`
+- Start Date: 2019-24-04
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Add functions to the standard library for testing a `char` against [UAX TR31](https://unicode.org/reports/tr31/) ("Unicode Annex 31")
+`Pattern_White_Space`, `Pattern_Syntax`, `XID_Start`, `ID_Nonstart`, and `XID_Continue` (the XID ones are already in the standard
+library, but are unstable; this RFC proposes to stablize them).
+
+# Motivation
+[motivation]: #motivation
+
+As a systems language, Rust is heavily used for parsing.
```

Comment:
To me this reads as motivation for why Rust needs such functions _somewhere_, but inclusion in the stdlib is a much higher bar, _especially_ when this gets us tangled up with unicode versioning.

Yes, we already have is_whitespace, but that's an old API that was grandfathered in, and it has fewer unicode stability issues than XID.

Reply:
One argument for it being in std is discoverability, but if you're parsing some grammar that grammar will likely tell you about XID.